[WIP] RHDEVDOCS-3819: [UG] How to store and retrieve logs (#60966)
openshift/openshift-docs: cicd/gitops/viewing-argo-cd-logs-using-the-kibana-dashboard.adoc

:_content-type: ASSEMBLY

> [id="viewing-argo-cd-logs-using-the-kibana-dashboard"]
> = Viewing Argo CD logs using the Kibana dashboard
> include::_attributes/common-attributes.adoc[]
> :context: viewing-argo-cd-logs-using-the-kibana-dashboard

Reviewer: I suggest removing the `using-the-kibana-dashboard` part:
```suggestion
[id="viewing-argo-cd-logs"]
= Viewing Argo CD logs
include::_attributes/common-attributes.adoc[]
:context: viewing-argo-cd-logs
```
Please note that it's also in the name of this file ^.

:_content-type: ASSEMBLY
[id="viewing-argo-cd-logs-using-the-kibana-dashboard"]
= Viewing Argo CD logs using the Kibana dashboard
include::_attributes/common-attributes.adoc[]
:context: viewing-argo-cd-logs-using-the-kibana-dashboard

toc::[]

> You can view the Argo CD logs with {logging-title}. The {logging} provides a visualization of logs using Kibana dashboard. To integrate Argo CD with {logging}, the default configuration for the OpenShift Logging Operator enables logging with Argo CD.

Reviewer: I'm not acquainted with this topic, so here is only a tentative suggestion for how to simplify this paragraph and add a few missing definitive articles:
```suggestion
You can view the Argo CD logs with the {logging-title}. The {logging} visualizes the logs on a Kibana dashboard. The OpenShift Logging Operator enables logging with Argo CD by default.
```

Reviewer 2: This entire paragraph of intro content should be moved from here into the procedure module .adoc file. The assembly file should look like this:

```
:_content-type: ASSEMBLY
[id="viewing-argo-cd-logs-using-the-kibana-dashboard"]
= Viewing Argo CD logs using the Kibana dashboard
include::_attributes/common-attributes.adoc[]
:context: viewing-argo-cd-logs-using-the-kibana-dashboard

toc::[]

include::modules/gitops-storing-and-retrieving-argo-cd-logs.adoc[leveloffset=+1]

[role="_additional-resources"]
[id="additional-resources_viewing-argo-cd-logs-with-the-kibana-dashboard"]
== Additional resources
* xref:../../logging/cluster-logging-deploying.adoc#cluster-logging-deploy-console_cluster-logging-deploying[Installing the {logging-title} using the web console]
```

[id="prerequisites_viewing-argo-cd-logs-with-the-kibana-dashboard"]
== Prerequisites

* The {gitops-title} Operator is installed in your cluster.

> * The {logging-title} is installed with default configuration  in your cluster.

Reviewer: I suggest condering moving these prerequisites section into `modules/gitops-storing-and-retrieving-argo-cd-logs.adoc` as `.Prerequisites` before `.Procedure`.

Reviewer 2: @max-cx is correct. The prereqs (plus the intro paragraph) should be moved to the procdedure module file.

include::modules/gitops-storing-and-retrieving-argo-cd-logs.adoc[leveloffset=+1]

[role="_additional-resources"]
[id="additional-resources_viewing-argo-cd-logs-with-the-kibana-dashboard"]
== Additional resources

> * xref:../../logging/cluster-logging-deploying.adoc#cluster-logging-deploy-console_cluster-logging-deploying[Installing the {logging-title} using the web console]

Reviewer: You can actually get rid of this completely by incorporating the link inline in the relevant prerequisite item. For example:
```
* The {logging-title} is installed with default configuration in your cluster. See xref:../../logging/cluster-logging-deploying.adoc#cluster-logging-deploy-console_cluster-logging-deploying[Installing the {logging-title} using the web console].
```
or
```
* The {logging-title} is xref:../../logging/cluster-logging-deploying.adoc#cluster-logging-deploy-console_cluster-logging-deploying[installed with default configuration] in your cluster.
```

Author: I agree with that approach. However, because we should not use xrefs in modules (as explained here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-procedures), I suggest leaving prerequisites in the assembly to include the xref as you suggested here. Thank you!
